include/rocksdb/write_batch.h

```diff
@@ -333,18 +336,32 @@ class WriteBatch : public WriteBatchBase {
   // Returns true if MarkRollback will be called during Iterate
   bool HasRollback() const;
 
+  // Experimental.
   // Assign timestamp to write batch.
-  // This requires that all keys (possibly from multiple column families) in
-  // the write batch have timestamps of the same format.
-  Status AssignTimestamp(const Slice& ts);
-
+  // This requires that all keys, if enable timestamp, (possibly from multiple
+  // column families) in the write batch have timestamps of the same format.
+  // checker: function object specified by caller to check the timestamp size
+  // of a column family.
+  template <typename Checker =
+                std::function<Status(uint32_t /*cf*/, size_t /*ts_sz*/)>>
```

Comment:
`std::function` does type erasure, which has a runtime overhead similar to a virtual method call. We could avoid that by introducing a small concrete class (e.g. `DummyChecker`), which would simply return `Status::OK()` from `operator()`. This class could then be used as the default template parameter

Reply:
Done